### PR TITLE
xds: add config for random and maglev load balancer extensions

### LINKED ISSUE #20634: xds: Envoy should support its built-in LB policies as extensions
LB extensibility was originally described in #5598.  The infrastructure for this has been implemented, but there are a few missing pieces.  Here's the relevant history:
- A new field for LB policy extensibility was added to xDS in #7744.
- The extension mechanism was added to Envoy in #17400.
- The semantics of the xDS field were changed slightly in #18419.
- xDS extension protos were added for ROUND_ROBIN and RING_HASH in #19517 and #20423, but Envoy does not yet actually implement configuring those policies as extensions.

I think the remaining work here is:
- Add extension protos for all built-in LB policies.
- Implement actually configuring these policies via the extension protos.

Once this is done, we can consider whether we want to deprecate the existing enum for configuring built-in policies.

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message: xds: add config for random and maglev load balancer extensions
Additional Description:

Part of work to close #20634. And this PR mark all extensions as not implemented. We should implement these extensions one by one.

Risk Level: low. no code change.
Testing: n/a.
Docs Changes: n/a.
Release Notes: n/a.
Platform Specific Features: n/a.

### CODE REVIEW

**Comment by kyessenov** [comment]
Should we also add one for ORIGINAL_DST?

**Comment by wbpcode** [comment]
> Should we also add one for ORIGINAL_DST?

The ORIGINAL_DST is only used by the `original dst cluster`. It's more like a `cluster_provider` lb config.

IMO, it would be better to make the `original dst cluster` as a custom cluster type in the future.

**Comment by wbpcode** [comment]
Hi, @markdroth , could you take a look again to the APIs?

File: api/envoy/extensions/load_balancing_policies/wrr_locality/v3/wrr_locality.proto
Comment (soulxu) [inline api/envoy/extensions/load_balancing_policies/wrr_locality/v3/wrr_locality.proto:21]: Should we hide this also?

File: api/envoy/extensions/load_balancing_policies/least_request/v3/least_request.proto
Comment (wbpcode) [inline api/envoy/extensions/load_balancing_policies/least_request/v3/least_request.proto:20]: file level hide is enabled. cc @soulxu

**Review by snowp — APPROVED** [review]
Thanks!

**Comment by wbpcode** [comment]
wait 1.24.0 release, then we can merge this pr.

**Comment by wbpcode** [comment]
I was hoping to merge this PR after 1.24.0 was released, but it looks like 1.24.0 will take a little longer. So maybe we can merge this API only PR now?

cc @phlax

**Comment by phlax** [comment]
> So maybe we can merge this API only PR now?

lets land it - it has approvals and shouldnt break anything i think

**Pull request merged**
